Dockerfile

```diff
@@ -19,27 +19,41 @@ RUN mkdir /src
 RUN adduser --disabled-password --gecos "" mitodl
 RUN mkdir /var/media && chown -R mitodl:mitodl /var/media
 
-# Poetry env configuration
+## Set some poetry config
 ENV  \
-  # poetry:
-  POETRY_VERSION=1.5.1 \
-  POETRY_VIRTUALENVS_CREATE=false \
-  POETRY_CACHE_DIR='/tmp/cache/poetry'
+  POETRY_VERSION=1.7.1 \
+  POETRY_VIRTUALENVS_CREATE=true \
+  POETRY_CACHE_DIR='/tmp/cache/poetry' \
+  POETRY_HOME='/home/mitodl/.local' \
+  VIRTUAL_ENV="/opt/venv"
+ENV PATH="$VIRTUAL_ENV/bin:$POETRY_HOME/bin:$PATH"
 
 # Install poetry
 RUN pip install "poetry==$POETRY_VERSION"
 
-# Install project packages
 COPY pyproject.toml /src
 COPY poetry.lock /src
-COPY mitol_django_*.gz /src
+RUN chown -R mitodl:mitodl /src
+RUN mkdir ${VIRTUAL_ENV} && chown -R mitodl:mitodl ${VIRTUAL_ENV}
+
+## Install poetry itself, and pre-create a venv with predictable name
+USER mitodl
+RUN curl -sSL https://install.python-poetry.org \
+  | \
+  POETRY_VERSION=${POETRY_VERSION} \
+  POETRY_HOME=${POETRY_HOME} \
+  python3 -q
 WORKDIR /src
+RUN python3 -m venv $VIRTUAL_ENV
 RUN poetry install
 
 # Add project
+USER root
 COPY . /src
 WORKDIR /src
-RUN chown -R mitodl:mitodl /src
+RUN mkdir /src/staticfiles
+
+RUN apt-get clean && apt-get purge
```

Comment:
Doesn't block merging but noting - this is here and also on line 12, prob don't need to clean up after apt twice